escrow/prepaid_service.go

```diff
@@ -0,0 +1,202 @@
+package escrow
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+)
+
+type lockingPrepaidService struct {
+	storage        *PrepaidStorage
+	validator      *PrePaidPaymentValidator
+	replicaGroupID func() ([32]byte, error)
+}
+
+func NewPrePaidService(
+	storage *PrepaidStorage,
+	prepaidValidator *PrePaidPaymentValidator, groupIdReader func() ([32]byte, error)) PrePaidService {
+	return &lockingPrepaidService{
+		storage:        storage,
+		validator:      prepaidValidator,
+		replicaGroupID: groupIdReader,
```

Comment:
these will come in the regualr flow in a different PR ( too much of code again , didn want to clutter this PR with too many files) its pretty big alreadt, once this one is cleared , it will be used by the prepaid validator